src/main/groovy/edu/oregonstate/mist/trailsapi/resources/TrailResources.groovy

```diff
@@ -99,7 +99,11 @@ public class TrailsResource extends Resource {
 			   @QueryParam("gap") Boolean gap) {
 		List<Trail> trails = trailDAO.getTrailByQuery(name, difficulty, mostDifficult, leastDifficult,
 			zipCode, smallDrop,largeDrop, woodRide, skinny, largeJump,smallJump, gap)
-		ok(trailResult(trails)).build()
+		if (trails) {
```

Comment:
I added the notFound response to the GET by query resource after realizing its usefulness in GET by id.